Makefile

```diff
@@ -126,15 +126,15 @@ distclean: clean
 install: | nvim
 	+$(BUILD_CMD) -C build install
 
-clint:
-	$(CMAKE_PRG) -DLINT_PRG=./src/clint.py \
-		-DLINT_DIR=src \
-		-DLINT_SUPPRESS_URL="$(DOC_DOWNLOAD_URL_BASE)$(CLINT_ERRORS_FILE_PATH)" \
-		-P cmake/RunLint.cmake
+clint: build/.ran-cmake
+	+$(BUILD_CMD) -C build clint
+
+clint-full: build/.ran-cmake
+	+$(BUILD_CMD) -C build clint-full
 
 check-single-includes: build/.ran-cmake
 	+$(BUILD_CMD) -C build check-single-includes
 
-lint: check-single-includes clint testlint
+lint: check-single-includes clint-full testlint
```

Comment:
`make lint` already does expensive `check-single-includes`. You do not run it if you care about time.

And by “may not check all files” I did *not* mean that it is going to check only new ones. I meant that once file X5 is checked and script determined that there are errors file X10 scheduled to be run 5 jobs next and also containing errors may be not checked at all because X5 already failed the build. This means developers will see less errors in CI.